boa_tester/src/main.rs

```diff
@@ -406,62 +563,48 @@ enum TestOutcomeResult {
     Panic,
 }
 
-#[derive(Debug, Serialize, Clone, Copy, Deserialize, PartialEq, Default)]
-#[serde(untagged)]
-enum SpecVersion {
-    ES5 = 5,
-    ES6 = 6,
-    #[default]
-    ES13 = 13,
-}
-
 /// Represents a test.
 #[derive(Debug, Clone)]
 #[allow(dead_code)]
 struct Test {
     name: Box<str>,
+    path: Box<Path>,
     description: Box<str>,
     esid: Option<Box<str>>,
-    spec_version: SpecVersion,
+    edition: SpecEdition,
     flags: TestFlags,
     information: Box<str>,
-    features: Box<[Box<str>]>,
     expected_outcome: Outcome,
-    includes: Box<[Box<str>]>,
+    features: FxHashSet<Box<str>>,
+    includes: FxHashSet<Box<str>>,
     locale: Locale,
-    path: Box<Path>,
     ignored: bool,
 }
 
 impl Test {
     /// Creates a new test.
-    fn new<N, C>(name: N, path: C, metadata: MetaData) -> Self
+    fn new<N, C>(name: N, path: C, metadata: MetaData) -> Result<Self>
     where
         N: Into<Box<str>>,
         C: Into<Box<Path>>,
     {
-        let spec_version = if metadata.es5id.is_some() {
-            SpecVersion::ES5
-        } else if metadata.es6id.is_some() {
-            SpecVersion::ES6
-        } else {
-            SpecVersion::ES13
-        };
+        let edition = SpecEdition::from_test_metadata(&metadata)
+            .ok_or_else(|| eyre!("couldn't find a minimum edition from the test metadata"))?;
```

Comment:
```suggestion
        let edition = SpecEdition::from_test_metadata(&metadata)
            .ok_or_else(|| eyre!("couldn't find a minimum edition from the test metadata"))?;
```
I think it would be nice to print the feature that was not found to `stdout`, so we can easily add it :)